src/main/java/com/bullhornsdk/data/api/helper/RestApiSession.java

```diff
@@ -289,7 +294,9 @@ public String getRestUrl() {
 	private synchronized void setBhRestToken(String bhRestToken) {
 		this.bhRestToken = bhRestToken;
 
-		updateDateTimeBhRestTokenWillExpire();
+		if(restCredentials.credentialCheck()) {
```

Comment:
I don't know if this is going to work properly - the logic is for when the bhRestToken is set immediately after getting it from the OAuth login process. If you take an existing one, how do you know how long it's been since the user last used it? This may take some play testing to ensure that the token expiring logic isn't messed up by providing an existing token from a separate login process.